src/features/editor/index.tsx

```diff
@@ -2,29 +2,43 @@
  * This file contains the Editor component which is used in Alignment Editor
  * mode and wraps Polyglot, ControlPanel and ContextPanel.
  */
-import { ReactElement } from 'react';
+import { ReactElement, useCallback, useState } from 'react';
 import { Container } from '@mui/material';
+import _ from 'lodash';
 
 import useDebug from 'hooks/useDebug';
 
 import Polyglot from 'features/polyglot';
 import ControlPanel from 'features/controlPanel';
 import ContextPanel from 'features/contextPanel';
 
-import { CorpusContainer } from 'structs';
+import { CorpusContainer, NamedContainers, Verse } from 'structs';
 
 import '../../styles/theme.css';
 import BCVWP from '../bcvwp/BCVWPSupport';
+import { AlignmentSide } from '../../common/data/project/corpus';
 
 interface EditorProps {
-  containers: CorpusContainer[];
+  containers: NamedContainers;
   position: BCVWP;
   usePaddingForEditorContainer?: boolean,
 }
 
-const Editor = ({containers, position, usePaddingForEditorContainer}: EditorProps): ReactElement => {
+const Editor = ({
+                  containers,
+                  position,
+                  usePaddingForEditorContainer
+                }: EditorProps): ReactElement => {
   useDebug('Editor');
 
+  const [visibleSourceVerses, setVisibleSourceVerses] = useState<Verse[]>([]);
+  const setNewVisibleVerses = useCallback((inputVerses: Verse[], corpus: CorpusContainer) => {
```

Comment:
Could you add a brief comment explaining what is happening inside this useCallback?

Reply:
Done.